src/main/java/org/swisspush/redisques/RedisQues.java

```diff
@@ -430,12 +430,12 @@ private void registerNotExpiredQueueCheck() {
             }
             String keysPattern = consumersPrefix + "*";
             log.debug("RedisQues list not expired consumers keys:");
-            redisProvider.redis().onSuccess(redisAPI -> redisAPI.keys(keysPattern, keysResult -> {
-                        if (keysResult.failed() || keysResult.result() == null) {
+            redisProvider.redis().onSuccess(redisAPI -> redisAPI.scan(Arrays.asList("0", "MATCH", keysPattern, "COUNT", "1000"), keysResult -> {
```

Comment:
I never used `scan` myself but don't you have to have to iterate over the whole keys by using the `cursor`?

Reply:
See screenshot, this is not needed

![image](https://github.com/user-attachments/assets/3893a3da-dd65-41fa-8ad6-924f968af97e)

Reply:
> See screenshot, this is not needed
> 
> ![image](https://private-user-images.githubusercontent.com/30789476/385116796-3893a3da-dd65-41fa-8ad6-924f968af97e.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzEzOTMwNTMsIm5iZiI6MTczMTM5Mjc1MywicGF0aCI6Ii8zMDc4OTQ3Ni8zODUxMTY3OTYtMzg5M2EzZGEtZGQ2NS00MWZhLThhZDYtOTI0Zjk2OGFmOTdlLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDExMTIlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMTEyVDA2MjU1M1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWYzYmNiNzUxYzU1ZDA2YzhhMjE2NWU5Y2M0MTI5ZTk1ODE4MTVhN2RlN2Q5ZWUxM2QzOTBiYjQ2NzFkMjY0OTEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.a5AfC_4qx3kDuXLiyE5GfezhhvYU93btWWbYj4S3_iQ)

I don't see why it is not needed. In the screenshot you were "lucky" to got "0" as cursor index result. When this is number higher than "0" you have to iterate again. The _SCAN_ command does not return all results, but at most the number you defined ("1000).